### DISCUSSION TITLE
Integrating custom device, MQTT over wifi

### DISCUSSION THREAD

**Discussion by pebblebed-tech** [Koenkk/zigbee2mqtt#10229] [Devices]
Hello
Background, I have a biomass plant that today is controlled by set of Arduino sensors and RASPI running a tiguitto stack with node.js webpage. Now I have decided to update the domestic heating system to become "Smart" eg TRV's etc. This then led me to your wonderful forum.

What I would like to do is use Z2M act as the single point of truth for both my custom wifi sensors and the zigbee bought in sensors. 
I have inspected the MQTT broker and I can see there are topics per device and a bridge topic which contains meta data about the devices.
If I just get my Arduino sensors/actuators to use the same MQTT broker as Z2M and post as per the device syntax, will I break / confuse your app if the sensors are not in the bridge topic? 
Also the plan would be to update my frontend to use Z2M api to access all data and control the actuators
Some advice would be appreciated

Thanks

**Comment by francisp2** [comment]
I use the same mqtt broker for zigbee2mqtt, tasmota, esprflink, openmqttgateway and some custom devices. All without conflict.

**Comment by pebblebed-tech** [comment]
@francisp2 
Thanks for the response.
Hi yes I know I can use the same broker with different top level topic branch. What I would like to do is add the custom non zigbee device into Z2MQTT topics so I can access all the devices via the Z2MQTT api rather than using the api for zibee and then MQTT client for the rest. Also this will give me a single engineering frontend to diagnose issues eg Z2MQTT frontend.
@Koenkk As the developer do you have any thoughts?

**Reply by Koenkk** [reply]
This is not possible since those devices will never be in the data/database.db and thus will be missing from zigbee2mqtt/bridge/devices topic which is used by the frontend.

**Reply by pebblebed-tech** [reply]
Ok thank you very much for the advice
